### PR TITLE
Updating loans_migrator task and legacy_loan class to use tenant timezone

### PR DESCRIPTION
**Pull request by bltravis**
I've updated the loans_migrator task to fetch the tenant's configured timezone and pass that to the legacy_loan class, rather than a simple offset value. This should enable the to_utc method to produce UTC timestamps that account for the DST.

As a bonus, I also fixed the "legacy_loan_count" off-by-one issue, as well. You can now post a loans file with only one loan in it.

### CODE REVIEW

**Comment by bltravis** [comment]
Some errors in the tests. I'm looking at them. Will update the pull request when I have a fix.

File: src/folio_migration_tools/migration_tasks/loans_migrator.py [hunk 1]
```diff
@@ -50,6 +52,23 @@ def __init__(
         task_configuration: TaskConfiguration,
         library_config: LibraryConfiguration,
     ):
+        try:
+            logging.info(
+                "Attempting to retrieve tenant timezone configuration..."
+            )
+            self.tenant_timezone_str = json.loads(self.folio_client.folio_get_single_object(
```
Comment (fontanka16) [inline src/folio_migration_tools/migration_tasks/loans_migrator.py:59]: Same here, the json.loads is likely superflous

File: src/folio_migration_tools/migration_tasks/loans_migrator.py [hunk 2]
```diff
@@ -50,6 +52,23 @@ def __init__(
...
+                "/configurations/entries?query=(module==ORG%20and%20configName==localeSettings)"
+                )["configs"][0]["value"])["timezone"]
+            logging.info(
+                "Tenant timezone is: %s", 
+                self.tenant_timezone_str
+            )
+        except Exception:
+            logging.info(
+                'Tenant locale settings not available. Using "UTC".'
```
Comment (fontanka16) [inline src/folio_migration_tools/migration_tasks/loans_migrator.py:68]: I wonder if this information (and the information of the tenant timezone) should be added to the migration report. 
But no need now.

File: src/folio_migration_tools/migration_tasks/loans_migrator.py [hunk 3]
```diff
@@ -256,10 +275,10 @@ def load_and_validate_legacy_loans(self, loans_reader):
             except ValueError as ve:
                 logging.exception(ve)
         logging.info(
-            f"Done validating {legacy_loan_count} legacy loans with {num_bad} rotten apples"
+            f"Done validating {legacy_loan_count + 1} legacy loans with {num_bad} rotten apples"
```
Comment (fontanka16) [inline src/folio_migration_tools/migration_tasks/loans_migrator.py:278]: Another way to fix this - that I actually prefer - is to set start=1 in the enumerate call. Never mind now though.

**Review by fontanka16 — APPROVED** [review]
Well done! An elegant solution.
Added two comments.

**Comment by bltravis** [comment]
@fontanka16 I'm going to try and run a couple more tests on both this and #288 today before I merge.

**Review by banerjek — APPROVED** [review]

**Review by fontanka16 — APPROVED** [review]

**Pull request merged**